### PR TITLE
Upgrade for tools to Scala 2.12.11, ScalaTest 3.1.1 and ScalaCheck 1.14.3.

### PR DESCRIPTION
**Pull request by ekrich**
Added a comment in the build as well.

### CODE REVIEW

**Comment by errikos** [comment]
I believe we should upgrade `org.scalamacros:paradise` to `2.1.1`. It seems that `2.1.0` is not (yet?) [published](https://mvnrepository.com/artifact/org.scalamacros/paradise) for 2.12.11.

**Comment by ekrich** [comment]
@errikos Oh gosh, I didn't run the tools test. I will go ahead and update that plus maybe some other dependencies too as part of our overall refresh.

File: tools/src/test/scala/scala/scalanative/nir/SigManglingSuite.scala [hunk 1]
```diff
@@ -3,7 +3,7 @@ package nir
 
 import org.scalatest._
 
-class SigManglingSuite extends FunSuite {
+class SigManglingSuite extends funsuite.AnyFunSuite {
```
Comment (sjrd) [inline tools/src/test/scala/scala/scalanative/nir/SigManglingSuite.scala:6]: In all other places you added an import for that. Could you do the same here, please?

**Comment by sjrd** [comment]
Anyone knows what we even use `paradise` for?

File: tools/src/test/scala/scala/scalanative/nir/SigManglingSuite.scala (hunk 1 above)
Comment (ekrich) [reply tools/src/test/scala/scala/scalanative/nir/SigManglingSuite.scala:6]: Sure, I originally did that and then went back so I must have missed that one.

**Comment by ekrich** [comment]
It looks like it may have only been needed for Scala 2.10 - https://github.com/lihaoyi/fastparse/issues/72

**Comment by sjrd** [comment]
> 
> 
> It looks like it may have only been needed for Scala 2.10 - [lihaoyi/fastparse#72](https://github.com/lihaoyi/fastparse/issues/72)

Do you want to try and get rid of it, while you're there, instead of upgrading it?

**Comment by ekrich** [comment]
Yes, please. If we don't need it I think that is best. I also would like to update fastparse but in another PR.

**Comment by ekrich** [comment]
@sjrd @errikos Thanks!

**Pull request merged**